shesha-reactjs/src/components/dataTable/index.tsx

```diff
@@ -693,6 +694,17 @@ export const DataTable: FC<Partial<IIndexTableProps>> = ({
     onResizedChange: onResizedChange,
   };
 
+  if (tableProps.columns && tableProps.columns.length === 0) {
+    return (
+      <Alert
+        showIcon
+        message="No Data"
+        description="No data is available for this table"
+        type="warning"
+      />
+    );
+  }
+
```

Comment:
Issue https://github.com/shesha-io/shesha-framework/issues/700 requires to add configuration but not hardcoded warning.
And also for DataList.

And I suggest to add configuration for Type (success, info, warning, error)